pkg/services/cosmosdb/common-provision.go

```diff
@@ -147,6 +210,35 @@ func (c *cosmosAccountManager) buildGoTemplateParams(
 	if len(filters) > 0 {
 		p["ipFilters"] = strings.Join(filters, ",")
 	}
+
+	if pp.ConsistencyPolicy != nil {
+		consistencyPolicy := make(map[string]interface{})
+		cp := strings.ToLower(pp.ConsistencyPolicy.DefaultConsistency)
+		switch cp {
+		case "eventual":
+			consistencyPolicy["defaultConsistencyLevel"] = "Eventual"
+		case "session":
+			consistencyPolicy["defaultConsistencyLevel"] = "Session"
+		case "boundedstaleness":
+			consistencyPolicy["defaultConsistencyLevel"] = "BoundedStaleness"
+		case "strong":
+			consistencyPolicy["defaultConsistencyLevel"] = "Strong"
+		case "consistentprefix":
+			consistencyPolicy["defaultConsistencyLevel"] = "ConsistentPrefix"
+		}
+
+		if pp.ConsistencyPolicy.MaxInternal != nil {
+			consistencyPolicy["maxIntervalInSeconds"] =
+				*pp.ConsistencyPolicy.MaxInternal
+		}
```

Comment:
Should there we a guard on this stanza to only set this if the policy is `BoundedStaleness`? Same question for the next stanza.

Reply:
Yeah, I misread the documentation. 

```